cmd/charm/charmcmd/push.go

```diff
@@ -110,10 +110,7 @@ func (c *pushCommand) Run(ctxt *cmd.Context) error {
 	defer client.jar.Save()
 
 	// Retrieve the source directory where the charm or bundle lives.
-	srcDir, err := filepath.Abs(c.srcDir)
-	if err != nil {
-		return errgo.Notef(err, "cannot access charm or bundle")
-	}
+	srcDir := ctxt.AbsPath(c.srcDir)
```

Comment:
This does not work as is (does not really return the absolute path, which we need), unless you change how the context is instantiated in cmd/charm/main.go. We must either pass an absolute path to Dir there or use filepath.Abs here.